modules/react-components/src/components/modal/edit-avatar-modal/edit-avatar-modal.tsx

```diff
@@ -283,14 +284,36 @@ export const EditAvatarModal: FunctionComponent<EditAvatarModalPropsInterface> =
         setGravatarURLs(urls);
     }, [ selectedGravatarEmail, isGravatarQualified ]);
 
+    /**
+     * Set the Avatar types based on imageUrl.
+     */
     useEffect(() => {
         if (imageUrl) {
-            setHostedURL(imageUrl);
-            setSelectedAvatarType(AvatarTypes.URL);
+            if (isGravatarUrl) {
+                setOutputURL(imageUrl);
+                setSelectedAvatarType(AvatarTypes.GRAVATAR);
+            } else {
+                setHostedURL(imageUrl);
+                setSelectedAvatarType(AvatarTypes.URL);
+            }
         } else {
             setSelectedAvatarType(AvatarTypes.SYSTEM_GENERATED);
         }
-    }, [ imageUrl ]);
+    }, [ imageUrl , isGravatarUrl ]);
+
+    /**
+     * Check the type of imageUrl.
+     */
+    useEffect(() => {
+
+        if (gravatarURLs) {
+            gravatarURLs.forEach((value: string, key: string) => {
+                if (imageUrl.localeCompare(value) == 0) {
+                    setIsGravatarUrl(true);
+                }
```

Comment:
I think we can break the loop if the `imageUrl` matches one of the URLs in the `gravatarURLs` array. Since `forEach` is a method and hence cannot be exited, we can use `for ... of`. 
```TypeScript
for(const value of gravatarURLs) {
    if (imageUrl.localeCompare(value) == 0) {                    
        setIsGravatarUrl(true);   

        break;
    }
}
```